### PR TITLE
Add rendered changes of programs check

### PR DESCRIPTION
**Pull request by ianyount**
Raise an error and reject build if if rendering causes changes within the `./programs` directory
Issue #24

### CODE REVIEW

File: incent-config/Dangerfile [hunk 1]
```diff
@@ -136,6 +136,14 @@ def strip_doc(doc)
   doc.strip.tr("\n", ' ')
 end
 
+def rendered_programs_changed?
+  `git status ./programs --short | wc -l`.strip != '0'
```
Comment (ianyount) [inline incent-config/Dangerfile:140]: This returns `false` if no changes are found in the `./programs` directory. Should we also add the check @jasmineHundal proposed of checking for a `./templates` folder as well or is that unneeded?

Comment (johnallen3d) [reply incent-config/Dangerfile:140]: I'm not 100% sure about this yet.

Comment (johnallen3d) [reply incent-config/Dangerfile:140]:
```sh
❯ git status --short
?? edge-retailer-2021/programs/foo.yml

…/incent-config-winfield on  feature/edge-retailer-2021/add-examples [?] 
❯ git status --short ./programs

…/incent-config-winfield on  feature/edge-retailer-2021/add-examples [?] 
❯ git status --short edge-retailer-2021/programs/
?? edge-retailer-2021/programs/foo.yml
```

Comment (johnallen3d) [reply incent-config/Dangerfile:140]: I like @jasmineHundal's idea about checking for a `templates` folder, however I'm not sure the added overhead of shelling out for that check is worth it. 🤔

**Review by johnallen3d — CHANGES_REQUESTED** [review]
See inline comments

**Review by johnallen3d — APPROVED** [review]

**Pull request merged**